barman/recovery_executor.py

```diff
@@ -1856,10 +1856,21 @@ def recover(self, backup_info, dest, remote_command=None, **kwargs):
         self.temp_dirs.append(fs.LocalLibPathDeletionCommand(synthetic_backup_dir))
 
         # Perform the standard recovery process passing the synthetic backup
-        return super(IncrementalRecoveryExecutor, self).recover(
+        recovery_info = super(IncrementalRecoveryExecutor, self).recover(
             synthetic_backup_info, dest, remote_command=remote_command, **kwargs
         )
 
+        # If the checksum configuration is not consistent among all backups in the chain, we
+        # raise a warning at the end so the user can optionally take action about it
+        if not backup_info.is_checksum_consistent():
+            output.warning(
+                "You recovered from an incremental backup where checksums were enabled on "
+                "some, but not all backups in the chain. It is advised to disable, and "
+                "optionally reenable, checksums on the destination directory to avoid failures."
```

Comment:
```suggestion
                "You recovered from an incremental backup where checksums were enabled on "
                "that backup, but not all backups in the chain. It is advised to disable, and "
                "optionally re-enable, checksums on the destination directory to avoid failures."
```

This sounds a bit "more accurate"?

Reply:
It does! The previous opens space for someone to think a backup with checksums disabled could also have its chain as inconsistent if any of the previous had it enabled, which is not true.